nntrainer/include/optimizer.h

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (C) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @file	optimizer.h
+ * @date	08 April 2020
+ * @brief	This is Optimizer classes of Neural Network
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+#ifndef __OPTIMIZER_H__
+#define __OPTIMIZER_H__
+
+#include <iostream>
+#include "tensor.h"
+
```

Comment:
Please consider applying namespace.

Reply:
And please prevent C code including this C++-only header. (use .hh as extension, add __cplusplus macro)

Reply:
I will apply them later. Thanks.